File: Creation/Managed/VB/2.multi-class/MultiClass.vb
' Create a managed DLL with Visual Basic.

' There will be three classes inside the DLL.
'     - AddClass
'     - MultiClass (this class)
'     - FactorialClass

' Compile:
'     $ vbc /t:library /out:managed.dll AddClass.vb MultiClass.vb FactorialClass.vb

imports System

' Questions:
'     - What do the DLL export?
'     - When reconstruct in .NET-aware decompiler, how much the decompiled code
'       resemblence the source code?
'     - When only this code compiled, what's the content of the DLL?

namespace CodeLibrary
    public class MultiClass
        public shared function Multiply(a as integer, b as integer) as integer
            return a * b
        end function
    end class
end namespace

File: Creation/Managed/C#/2.multi-class/MultiClass.cs
/*
Create a managed DLL with C#.

There will be three classes inside the DLL.
    - AddClass
    - MultiClass (this class)
    - FactorialClass

Compile:
    $ csc /t:library /out:managed.dll AddClass.cs MultiClass.cs FactorialClass.cs
*/

using System;

/*
Questions:
    - What do the DLL export?
    - When reconstruct in .NET-aware decompiler, how much the decompiled code
      resemblence the source code?
    - When only this code compiled, what's the content of the DLL?
*/

namespace CodeLibrary
{
    public class MultiClass
    {
        public static int Multiply(int a, int b)
        {
            return a * b;
        }
    }
}